pkgs/sops-install-secrets/main.go

```diff
@@ -102,6 +103,28 @@ type appContext struct {
 	checkMode   CheckMode
 }
 
+func secureSymlinkChown(symlinkToCheck, expectedTarget string, owner, group int) error {
+	fd, err := unix.Open(symlinkToCheck, unix.O_CLOEXEC|unix.O_PATH|unix.O_NOFOLLOW, 0)
+	if err != nil {
+			return fmt.Errorf("Failed to open %s: %w", symlinkToCheck, err)
+	}
+	defer unix.Close(fd)
+
+	buf := make([]byte, len(expectedTarget) + 1) // oversize by one to detect trunc
+	n, err := unix.Readlinkat(fd, "", buf)
+	if err != nil {
+		return fmt.Errorf("couldn't readlinkat %s", symlinkToCheck)
+	}
+	if n > len(expectedTarget) || string(buf[:n]) != expectedTarget  {
+		return fmt.Errorf("symlink %s does not point to %s", symlinkToCheck, expectedTarget)
+	}
+	err = unix.Fchownat(fd, "", owner, group, unix.AT_EMPTY_PATH|unix.AT_SYMLINK_NOFOLLOW)
```

Comment:
Since our file descriptor already points to right inode, we can skip `AT_SYMLINK_NOFOLLOW` here:

```suggestion
	err = unix.Fchownat(fd, "", owner, group, unix.AT_EMPTY_PATH)
```

It does not make a function difference here but at least make it clear that we understood the manpage.